---
arturl_encode: "68747470733a2f2f626c6f67:2e6373646e2e6e65742f77656978696e5f3339393733383130:2f61727469636c652f64657461696c732f3930383133393435"
layout: post
title: "fastdfs-java.net.SocketTimeoutException-connect-timed-out解决办法"
date: 2025-01-03 13:53:52 +0800
description: "错误信息java.net.SocketTimeoutException: c"
keywords: "spirngboot链接fastdfs提示connect timed out"
categories: ['学成在线', 'Springboot', 'Fastdfs']
tags: ['学成在线', 'Springboot', 'Fastdfs']
artid: "90813945"
image:
    path: https://api.vvhan.com/api/bing?rand=sj&artid=90813945
    alt: "fastdfs-java.net.SocketTimeoutException-connect-timed-out解决办法"
render_with_liquid: false
featuredImage: https://bing.ee123.net/img/rand?artid=90813945
featuredImagePreview: https://bing.ee123.net/img/rand?artid=90813945
---

# fastdfs java.net.SocketTimeoutException: connect timed out解决办法

错误信息

```java
java.net.SocketTimeoutException: connect timed out
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at org.csource.fastdfs.ClientGlobal.getSocket(ClientGlobal.java:208)
	at org.csource.fastdfs.StorageServer.<init>(StorageServer.java:43)
	at org.csource.fastdfs.TrackerClient.getStoreStorage(TrackerClient.java:144)
	at org.csource.fastdfs.TrackerClient.getStoreStorage(TrackerClient.java:68)
	at com.xuecheng.test.fastdfs.TestFastDFS.testUpload(TestFastDFS.java:31)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.springframework.test.context.junit4.statements.RunBeforeTestExecutionCallbacks.evaluate(RunBeforeTestExecutionCallbacks.java:73)
	at org.springframework.test.context.junit4.statements.RunAfterTestExecutionCallbacks.evaluate(RunAfterTestExecutionCallbacks.java:83)
	at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:75)
	at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:86)
	at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:84)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:251)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:97)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:190)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
2019-06-04 21:36:11.929  INFO 9443 --- [       Thread-2] o.s.w.c.s.GenericWebApplicationContext   : Closing org.springframework.web.context.support.GenericWebApplicationContext@32502377: startup date [Tue Jun 04 21:36:04 CST 2019]; root of context hierarchy

Process finished with exit code 0

```

网上找到的问题原因说是因为storage.conf配置文件中的端口22122没有放开, 但我确实已经在阿里云放开该端口了啊, 而且我Cenos7的防火墙也已经关闭了
  
![在这里插入图片描述](https://i-blog.csdnimg.cn/blog_migrate/9fc6e2504687c0e241972d4b29ca4658.png)
  
后来找了又找一了下发现需要将storage.conf配置文件中所以涉及到
**所有端口**
都要对外放开,查找一下storage.conf中另一个端口信息
  
![](https://i-blog.csdnimg.cn/blog_migrate/9253a4c29761859220bb009f9afdbb21.png)

于是阿里云放开这两个端口
  
![在这里插入图片描述](https://i-blog.csdnimg.cn/blog_migrate/51a99224f291133130505d3c7cec6bc6.png)
  
test测试问题解决